soh/soh/Enhancements/randomizer/draw.cpp

```diff
@@ -50,14 +77,28 @@ extern "C" void Randomizer_DrawSmallKey(PlayState* play, GetItemEntry* getItemEn
     gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx, (char*)__FILE__, __LINE__),
               G_MTX_MODELVIEW | G_MTX_LOAD);
 
-    if (isColoredKeysEnabled) {
-        gDPSetGrayscaleColor(POLY_OPA_DISP++, colors[color_slot][0], colors[color_slot][1], colors[color_slot][2], 255);
-        gSPGrayscale(POLY_OPA_DISP++, true);
-    }
+    Color_RGB8 keyColor = { 255, 255, 255 };
+    keyColor = CVarGetColor24(SmallBaseCvarValue[slot], keyColor);
+
+    if (isCustomKeysEnabled) {
+        gDPSetEnvColor(POLY_OPA_DISP++, keyColor.r, keyColor.g, keyColor.b, 255);
+        gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gSmallKeyCustomDL);
+
+        Gfx_SetupDL_25Xlu(play->state.gfxCtx);
+
+        Color_RGB8 iconColor = { 255, 0, 0 };
+        iconColor = CVarGetColor24(SmallIconCvarValue[slot], iconColor);
 
-    gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gGiSmallKeyDL);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx, (char*)__FILE__, __LINE__),
+                G_MTX_MODELVIEW | G_MTX_LOAD);
+        gDPSetEnvColor(POLY_XLU_DISP++, iconColor.r, iconColor.g, iconColor.b, 255);
+
+        gSPDisplayList(POLY_XLU_DISP++, CustomdLists[slot]);
 
-    if (isColoredKeysEnabled) {
+    } else {
+        gDPSetGrayscaleColor(POLY_OPA_DISP++, keyColor.r, keyColor.g, keyColor.b, 255);
+        gSPGrayscale(POLY_OPA_DISP++, true);
+        gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gGiSmallKeyDL);
```

Comment:
If I understand correctly it's no longer possible to use the vanilla models with colors, I'm not sure if this is something important, more people should weigh in.

Reply:
You misunderstand the code because the diff is being bad. What I did was move color handling to the cosmetic editor, and change the coloured keys checkbox to a custom key checkbox, which enables the Dana models. The gDPSetGrayscaleColor line still sets the colour even on this branch. 

I did this so the same colour changing cvars are used for both regular and Dana models.